src/other/utils.jl

```diff
@@ -168,3 +168,11 @@ function _nt_like_hash(v, h::UInt)
 
     return xor(objectid(Tuple(propertynames(v))), h)
 end
+
+function _findall(v::AbstractVector{Bool})
+    if all(v)
```

Comment:
this is not what I have suggested.

The algorithm I think we should investigate is:
1. for safety reasons check that `firstindex(v) == 1` (which should always be the case)
2. locate first `true` in `v`
3. locate end of continuous block of `true` in `v` since start
4. check if there exists any `true` beyond this end identified in point 2

Now we have three scenarios:
1. the block of `true` is continuous and spans from `1` to `length(v)` - then return `Base.OneTo(length(v))`.
2. the block of `true` is continuous but has a shorter span then return `start:stop` of the block (identified above)
3. otherwise materialize `start:stop` range and start adding to it `trues` identified later (for this first finish doing `count(v)` from the point where we have started and then reuse the code from `findall` but taking into account the fact that we should not re-do work already done)

Is the algorithm clear?

(of course like with `completecases` caution must be taken to make sure that the operation we make is fast as maybe some optimizations can be made - e.g. if we want to use `_findall` only for `BitVector` maybe it will be better to process the array in chunks taking advantage of the storage mode of `BitVector` - this would have to be benchmarked)

If anything of what I write above is not clear please comment.

In general the point is that e.g. in your code you call `all(v)` and later `findall` will call `count(v)` anyway, so it is better to do `count(v)` and compare it to the length of `v` rather than redoing the work (what I propose is even more complex but I make this comment to show you the way we should think while designing the algorithm - avoid doing the same work twice).

Reply:
Just to be clear what I mean by taking advantage of low level storage of `BitVector`. If you write:
```
julia> x = trues(1000);

julia> x.chunks
16-element Vector{UInt64}:
 0xffffffffffffffff
 0xffffffffffffffff
 0xffffffffffffffff
 0xffffffffffffffff
 0xffffffffffffffff
 0xffffffffffffffff
 0xffffffffffffffff
 0xffffffffffffffff
 0xffffffffffffffff
 0xffffffffffffffff
 0xffffffffffffffff
 0xffffffffffffffff
 0xffffffffffffffff
 0xffffffffffffffff
 0xffffffffffffffff
 0x000000ffffffffff
```
you can super fast identify continuous blocks of `true`s or `false`s as you have 64 element (or 32 element on 32-bit machine) blocks that are very easy and fast to check. All falses are just `0` and all `trues` are `typemax`. (care should be taken with the first and last block in the continuous range which can be partially filled, but again - it should be easy to check using bit operations)

Reply:
Hmm, ok, now I get it, you meant fully custom `findall`.
I thought about `all(v)` as it's short-circuiting i.e. returns on the first encountered false so it's not that bad.
I can play with doing this optimizations, let's see what they bring.

Reply:
In general up to 1.0 release we mostly focused the development effort on API design in DataFrames.jl.

In post 1.0 development performance will take more priority. Therefore I would prefer to work on new implementations of functions in detail (so that we can have the best design we can think of and not have to come back to it later).

Reply:
Also - my approach does not have to be best. If you can come up with something else that is good then of course please go ahead and propose something. We do not need to rush with the changes.

Reply:
As a reference I have just checked how `all` is implemented for `BitArray` and it uses the approach using `chunks` I suggested above.

Reply:
I'm working on this custom `findall` function.
IINM depending on the case we want to return one of `Base.oneTo{Int64}`, `UnitRange{Int64}`, `Vector{Int64}`, right?

Reply:
Thinking about it now probably `UnitRange{Int}` and `Vector{Int}` should be enough. The performance benefit of `Base.OneTo{Int}` is probably negligible (if any) and it is better to have a smaller union (2 types) in return type, as it will be better handled by the compiler.

Reply:
Sure thing, thanks